### ISSUE TITLE
[Bug] Path aliases easily confused with package scopes

### ISSUE THREAD

**Issue by joebowbeer** [temporalio/sdk-node#78]
**Describe the bug**
The use of `@` aliases in the auto-generated packages is unnecessarily confusing and may increase the chance of unintended aliasing with 3rd party packages:

```
      paths: {
        '@workflows': ['../workflows'],
        '@workflows/*': ['../workflows/*'],
        '@activities': ['../activities'],
        '@activities/*': ['../activities/*'],
        '@interfaces': ['../interfaces'],
        '@interfaces/*': ['../interfaces/*'],
      },
```

Source: https://github.com/temporalio/sdk-node/blob/main/packages/create-project/src/index.ts#L128

For many developers and code reviewers, the `@` prefix indicates a package scope, such as `@temporalio` and not an alias prefix.

At the least, I suggest changing to a different prefix that won't be confused with similarly-spelled package scopes. 

However, because of the many problems and surprises with path aliases (see below), my advice is to _not_ add them to auto-generated code. Rather, let the devs add them at their discretion.

**Additional context**
In my experience, Typescript's path aliases have been a continual source of problems. I do use them on large projects, but I've made certain compromises in doing so, and always with the fear that there is yet another issue to be uncovered.

For example:

https://mitchellsimoens.com/2019/08/07/why-typescript-paths-failed-me/

**Comment by bergundy** [comment]
Hi,

I'm totally fine with changing the prefix from `@`, do you have a suggestion for a better one?
I've only seen `@` used for aliases before and FWIU that's what is commonly used.

The article has some really good points but they don't apply to us since we only use the aliases for type information meaning that the aliases are omitted from the generated code.
There's one thing I would change in the in package initializer and that's not to add aliases to the worker package.
The only aliases we should have is `@activities` and `@interfaces` in the workflow project and `@interfaces` in the activities package.
WDYT?

**Comment by joebowbeer** [comment]
`~` is another common alias character. I'd prefer it to `@`.

The problems I have path aliases stem from the fact that they are designed for loaders that allow remapping, not for rewriting paths:

https://github.com/microsoft/TypeScript/issues/10866#issuecomment-246789510

**Comment by bergundy** [comment]
I'm fine with changing to `~` (even though it's also an alias to a user's home directory).

We have a custom module loader for the isolate runtime so that qualifies as a valid use case for path aliases.
See the worker code for loading activities [here](https://github.com/temporalio/sdk-node/blob/8ccedd2d4f186a5bc50019caea0bba7480dbc106/packages/worker/src/worker.ts#L387).
As for the `@interfaces` alias, it's supposed to only hold type definitions so it should be OK.

Does that make sense to you?

**Comment by joebowbeer** [comment]
Thanks for pointing me to the loader code.

A potential issue with an `interfaces` path alias might be if they are published and later imported. I think the path aliases would need to be rewritten before publication, both in the published code and in the published ts types.

**Comment by bergundy** [comment]
That's a fair point.
Can you try and reproduce this behavior?

**Comment by joebowbeer** [comment]
> Can you try and reproduce this behavior?

I think that's the first issue described here:

https://mitchellsimoens.com/2019/08/07/why-typescript-paths-failed-me/

To repro, look at the generated code and `types.d.ts` output from `npm pack`

I've used `ttsc` to translate the path aliases before publication.

**Comment by bergundy** [comment]
Just reproduced it, thanks!
The aliases are definitely broken after `npm pack`.
We can get rid of the interfaces alias but we need the activities alias for the module loader.
I suggest we either change the behavior of the module loader (not sure how yet) or live with the restriction that the project cannot be published.

**Comment by bergundy** [comment]
Or use ttsc as you've pointed out to keep the current behavior and support publishing.

**Comment by bergundy** [comment]
Once #106 is merged I think we can close this issue.
The only alias left is the special `@activities` alias which is required for our custom webpack resolver (https://github.com/temporalio/sdk-node/blob/d59051c732e961100ba75fdc431b742a489cfebb/packages/worker/src/isolate-builder.ts#L93).
I'm OK with the `@` prefix and I believe it's not that confusing and is more commonly used than the alternative proposed here (`~`), we can easily make this configurable BTW.

If you feel strongly against this decision feel free to reopen the issue and continue the discussion.
